### DISCUSSION TITLE
Ramping constraints

### DISCUSSION THREAD

**Discussion by r0sv** [spine-tools/SpineOpt.jl#832] [Support]
Hello!
Sorry, maybe I'm wrong because I have little experience using SpineOpt.
After successfully implementing the example of a simple system and the example from the webinar, I tried to apply Ramping constraints.
https://spine-tools.github.io/SpineOpt.jl/latest/tutorial/ramping/
unit_capacity =100
start_up_limit = 0.1
initial_unit_flow=0
But the power plant is turned on at full load (100 MW) without a gradual increase in power (e.g. 10 MW).
In addition, the constraints "ramp_up_limit" or "rump_down_limit" are not executed at all. The simulation ends with an error.
I have upgraded spineopt. I'm using the latest templates (spineopt_template and basic_model_template). It did not help.
Please, help me to understand how to use ramping constraints.

**Comment by datejada** [comment]
Hi,

Sorry for the delay; all of the developers were on holiday 🎄 😃 

Regarding your question, could you upload your input data file as JSON? You can do that by clicking on the export option in the Spine DB Editor, as shown in the following screenshots. Then, I can run it locally and see the problem easily.

thanks!

<img width="960" alt="image" src="https://github.com/spine-tools/SpineOpt.jl/assets/12887482/418542ff-a36c-4ca4-85d6-559ed6a2d8ab">

<img width="258" alt="image" src="https://github.com/spine-tools/SpineOpt.jl/assets/12887482/a37ed1d9-2e3a-4d49-bbf0-42bc6770e034">


<img width="473" alt="image" src="https://github.com/spine-tools/SpineOpt.jl/assets/12887482/897b2f9d-3af7-4104-911b-22a5fbc97e19">

**Reply by r0sv** [reply]
Hello and thank you for reply!
[Ramp.json](https://github.com/spine-tools/SpineOpt.jl/files/13820588/Ramp.json)

Over the past time, I have partially managed to solve the problem.
1. In the original situation, I installed SpineOpt on 4 computers using technique "from the Spine Julia Registry" (step 3a in documentation (https://github.com/spine-tools/SpineOpt.jl), without installation of SpineInterface). Next, when creating a project, I imported two files: spineopt_template.json and basic_model_template.json (these two files are mentioned in tutorial "Simple system" https://spine-tools.github.io/SpineOpt.jl/latest/tutorial/simple_system/)
After that I created a project with ramping constraints like the one I uploaded in this letter. But the constraints were not taken into account. Also I upgdared SpinOpt using:
using Pkg
pkg"up SpineOpt"
It didn't help.
2. After writing my question here, I installed SpineOpt to another computer using technique "from sources" (step 3b in documentation...I cloned SpineOpt, SpineInterface and so on (all points from 3b))
In addition, I imported other files: spineopt_template.json and simple_system.json (instead of basic_model_template.json)
And my Project (attached file - Ramp.json) began to take into account ramping constraints.
But this Project still doesn't work on other computers where SpineOpt is installed "from the Spine Julia Registry".
Therefore, now the questions are related with this issue. I want to understand what helped in solving the problem: installing the Spine Interface or changing the installation method or something else...

**Reply by datejada** [reply]
@r0sv Thanks for the detailed reply. So, to answer your latest question, you need to install both SpineOpt and SpineInterface to work properly. So, please follow the instructions in 3b on all your computers for future installations.

**Comment by clizbe** [comment]
@r0sv Did you figure it out? (We're working on improving our install instructions if that was the problem.)
